src/Steps/Release/MergeUpRelease.php

```diff
@@ -0,0 +1,201 @@
+<?php
+
+namespace SilverStripe\Cow\Steps\Release;
+
+use Gitonomy\Git\Exception\ProcessException;
+use SilverStripe\Cow\Model\Release\LibraryRelease;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+
+/**
+ * Merge up released commits to the major branch and optionally remove x.y-release branch.
+ * E.g. if you just released 4.12.0, this will merge 4.12-release up to 4.12, and then 4.12
+ * up to 4.
+ */
+class MergeUpRelease extends ReleaseStep
+{
+    public function getStepName()
+    {
+        return 'mergeup';
+    }
+
+    public function run(InputInterface $input, OutputInterface $output)
+    {
+        $this->log($output, "Merging up all modules");
+
+        $this->mergeRecursive(
+            $output,
+            $this->getReleasePlan(),
+            !$input->getOption('no-push')
+        );
+
+        $this->log($output, "All modules merged up");
+    }
+
+    /**
+     * Merge a library and its children up to their major branches
+     *
+     * @param OutputInterface $output
+     * @param LibraryRelease $releasePlanNode Node in release plan being released
+     */
+    protected function mergeRecursive(
+        OutputInterface $output,
+        LibraryRelease $releasePlanNode,
+        bool $pushToRemote
+    ) {
+        $currentBranch = $releasePlanNode->getLibrary()->getBranch();
+
+        // Skip upgrade-only modules and anything not on an actual branch (e.g. using a pre-existing tag)
+        if ($releasePlanNode->getLibrary()->isUpgradeOnly() || !$currentBranch) {
+            return;
+        }
+        $currentLibrary = $releasePlanNode->getLibrary()->getName();
+
+        // Skip if it's not a proper release branch.
+        if (!$releasePlanNode->isOnReleaseBranch()) {
+            $this->log(
+                $output,
+                "Library $currentLibrary is on a non-release branch '$currentBranch'"
+                    . "Please checkout the release branch and then run the command again.",
+                'error'
+            );
+            die();
+        }
+
+        // Do the merge up recursively
+        foreach ($releasePlanNode->getItems() as $item) {
+            $this->mergeRecursive($output, $item, $pushToRemote);
+        }
+        $this->mergeLibrary($output, $releasePlanNode, $pushToRemote);
+    }
+
+    /**
+     * Performs a merge up for a single library
+     *
+     * @param OutputInterface $output
+     * @param LibraryRelease $releasePlanNode Node in release plan being released
+     */
+    protected function mergeLibrary(OutputInterface $output, LibraryRelease $releasePlanNode, bool $pushToRemote)
+    {
+        $library = $releasePlanNode->getLibrary();
+        $releaseBranch = $library->getBranch();
+        $minorBranch = str_replace(LibraryRelease::RELEASE_BRANCH_SUFFIX, '', $releaseBranch);
+        $majorBranch = $releasePlanNode->getVersion()->getMajor();
```

Comment:
I can't see any value in doing that because a) we'll never be in that situation (how would we have a `4.12-release` branch without a corresponding `4.12` and `4` branch?) and b) even if we somehow _did_ get in that situation, cow will find that when it tries to checkout the branches, and halt at that stage.

I'll add the checks if you insist, but I don't think it's necessary.

Reply:
My main concern would be if $releaseBranch ($library->getBranch()) wasn't actually a release branch because it was manually checked out by someone for whatever reason

Reply:
That's already protected against in `mergeRecursive()` - but I'll add explicit checks so there's no ambiguity.

Reply:
Talked about offline, there should be no way that the branches are not what's expected due to validation that we're on a release branch on the previous method that calls this method